### PR TITLE
[serverless] Correct cli.log signature

### PR DESCRIPTION
**Pull request by AbdullahAli**
See correct usage here: https://github.com/serverless/serverless/blob/01f158695b22d721320a77a9a0b68b166c63dc3f/lib/classes/ConfigSchemaHandler/index.js#L82

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/serverless/serverless/blob/01f158695b22d721320a77a9a0b68b166c63dc3f/lib/classes/ConfigSchemaHandler/index.js#L82

### CODE REVIEW

File: types/serverless/serverless-tests.ts [hunk 1]
```diff
@@ -73,6 +73,18 @@ class BadVariablePlugin implements Plugin {
...
+serverless.cli.log(); // $ExpectError
+
+// Test serverless cli log with no entity
+serverless.cli.log('updating stack...');
+
+// Test serverless cli log with no options
+serverless.cli.log('updating stack...', 'serverless');
+
+// Test serverless cli log with all args supplied
+serverless.cli.log('updating stack...', 'serverless');
```
Comment (G-Rath) [inline types/serverless/serverless-tests.ts:86]:
```suggestion
serverless.cli.log('updating stack...', 'serverless', { color: 'orange' });
```

Currently this doesn't actually have all the args 😅

File: types/serverless/index.d.ts [hunk 2]
```diff
@@ -77,7 +78,7 @@ declare class Serverless {
     getVersion(): string;
 
     cli: {
-        log(message: string): null;
+        log(message: string, entity?: string, options?: object): null;
```
Comment (G-Rath) [inline types/serverless/index.d.ts:81]:
```suggestion
        log(message: string, entity?: string, options?: { color?: string }): null;
```

We know that `color` is at least one of the supported options - I've not actually found any other options glancing over the codebase.

**Review by G-Rath — CHANGES_REQUESTED** [review]
Two minor changes that I think would be good to make :)

**Comment by AbdullahAli** [comment]
Thank you @G-Rath good idea 👍🏼 
I dug into the code and managed to find some more options, this is how it is set up initially:

https://github.com/serverless/serverless/blob/master/lib/classes/CLI.js#L72

which makes use of a package called `@serverless/utils/log`: https://github.com/serverless/utils/blob/HEAD/docs/log.md

I've added the extra options and test them locally, with `entity`, `color`, `underline` and `bold`
Screenshot 2021-07-04 at 10 26 58

File: types/serverless/index.d.ts [hunk 3]
```diff
@@ -77,7 +78,7 @@ declare class Serverless {
     getVersion(): string;
 
     cli: {
-        log(message: string): null;
+        log(message: string, entity?: string, options?: { color?: string, bold?: boolean, underline?: boolean, entity?: string }): null;
```
Comment (G-Rath) [inline types/serverless/index.d.ts:81]: Given its size, I think it would be best to pull this out into a `LogOptions` interface

**Review by G-Rath — APPROVED** [review]
Awesome, looks good - thanks for finding all those extra properties.

I've made a comment about creating an interface for the `options` type, which'd be great if you could action but I don't think is a direct blocker.

File: types/serverless/index.d.ts (hunk 3 above)
Comment (AbdullahAli) [reply types/serverless/index.d.ts]: Great idea 👍🏼 I've added the `LogOptions` interface as you suggested [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54264/commits/8eb67cb96c00850245395ea30adfa2a913682948)

**Review by G-Rath — APPROVED** [review]

**Comment by AbdullahAli** [comment]
Ready to merge

**Pull request merged**